materialize-sql/std_sql.go

```diff
@@ -468,6 +468,94 @@ func StdCreateSchema(ctx context.Context, db *sql.DB, dialect Dialect, schemaNam
 	return err
 }
 
+type ColumnMigrationStep func(dialect Dialect, table Table, migration ColumnTypeMigration, firstTempColumn, secondTempColumn string) (string, error)
+
+var StdMigrationSteps = []ColumnMigrationStep{
+	func(dialect Dialect, table Table, migration ColumnTypeMigration, firstTempColumnIdentifier, _ string) (string, error) {
+		return fmt.Sprintf("ALTER TABLE %s ADD COLUMN %s %s;",
+			table.Identifier,
+			firstTempColumnIdentifier,
+			migration.DDL,
+		), nil
+	},
+	func(dialect Dialect, table Table, migration ColumnTypeMigration, firstTempColumnIdentifier, _ string) (string, error) {
+		return fmt.Sprintf(
+			// The WHERE filter is required by some warehouses (bigquery)
+			"UPDATE %s SET %s = CAST(%s AS %s) WHERE true;",
+			table.Identifier,
+			firstTempColumnIdentifier,
+			migration.Identifier,
+			migration.DDL,
+		), nil
+	},
+	func(dialect Dialect, table Table, migration ColumnTypeMigration, _, secondTempColumnIdentifier string) (string, error) {
+		return fmt.Sprintf(
```

Comment:
@williamhbaker that's an interesting proposal. I put in the renaming as a safety mechanism, but I see your point that it may not actually be as useful... We intend to remove the column anyway, and by removing that rename, we simplify MySQL's code a lot as well, since right now we need to reconstruct the original column's DDL (which we can't do accurately... and may cause issues).

I think I'm onboard with removing that extra rename step, I'll make the change 👍🏽